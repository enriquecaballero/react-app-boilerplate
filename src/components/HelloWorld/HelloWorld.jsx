import React from "react";
import StoreRenderer from "../StoreRenderer";
import styles from "./HelloWorld.scss";

export default function HelloWorld(props) {
  return (
    <div>
      <h2 className={styles.Title}>Hello, {props.name}!</h2>
      <input
        value={props.name}
        onChange={event => props.setName(event.target.value)}
      />
      <StoreRenderer />
    </div>
  );
}

HelloWorld.defaultProps = {
  name: ""
};
