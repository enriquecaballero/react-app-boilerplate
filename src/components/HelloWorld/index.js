import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import HelloWorld from "./HelloWorld";
import { setName } from "../../reducers/name";

function mapStateToProps (state): Object {
  return {
    name: state.name
  };
}

function mapDispatchToProps (dispatch): Object {
  return bindActionCreators (
    {
      setName
    },
    dispatch
  );
}

export default connect (mapStateToProps, mapDispatchToProps) (HelloWorld);
