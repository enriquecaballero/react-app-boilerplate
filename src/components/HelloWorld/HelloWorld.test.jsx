import React from "react";
import renderer from "react-test-renderer";
import HelloWorld from "./HelloWorld";

jest.mock("../StoreRenderer", () => "StoreRenderer");

describe("HelloWorld", () => {
  it("renders", () => {
    const component = renderer.create(<HelloWorld name="Bob" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
