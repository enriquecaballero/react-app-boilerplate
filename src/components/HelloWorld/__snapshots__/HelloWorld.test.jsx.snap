// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld renders 1`] = `
<div>
  <h2
    className="Title"
  >
    Hello, 
    Bob
    !
  </h2>
  <input
    onChange={[Function]}
    value="Bob"
  />
  <StoreRenderer />
</div>
`;
